pybv/io.py

```diff
@@ -133,14 +129,27 @@ def write_brainvision(*, data, sfreq, ch_names, fname_base, folder_out,
 
     _chk_fmt(fmt)
 
-    if unit == 'μV':
-        # this is greek mu: μ
-        # https://www.compart.com/de/unicode/U+03BC
-        warnings.warn(
-            f"Encountered small greek letter mu: 'μ' in unit: {unit} ... "
-            f"converting to micro sign: 'µ': {unit.replace('μ', 'µ')}"
-        )
-        unit = 'µV'
+    # check unit is single str
+    if isinstance(unit, str):
+        # convert unit to list, assuming all units are the same
+        unit = [unit] * nchan
+    if len(unit) != nchan:
+        raise ValueError(f"Number of channels in unit ({len(unit)}) "
+                         f"does not match number of channel "
+                         f"names ({nchan})")
+    units = unit
+
+    # check units for compatibility with greek lettering
+    for idx, unit in enumerate(units):
+        if unit == 'μV':
+            # this is greek mu: μ
+            # https://www.compart.com/de/unicode/U+03BC
+            warnings.warn(
+                f"Encountered small greek letter mu: 'μ' in unit: {unit} ... "
+                f"converting to micro sign: 'µ': {unit.replace('μ', 'µ')}"
+            )
```

Comment:
IMHO, this warning should be sent once in that loop. I wouldn't want to see this warning nchans times as a user. Especially not if I originally just passed a single unit :-)